pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/credentials/impl/StringCredentialsHandler.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2016-2018, CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.jenkinsci.plugins.pipeline.modeldefinition.credentials.impl;
+
+import com.cloudbees.plugins.credentials.common.StandardCredentials;
+import hudson.Extension;
+import org.jenkinsci.plugins.credentialsbinding.MultiBinding;
+import org.jenkinsci.plugins.credentialsbinding.impl.StringBinding;
+import org.jenkinsci.plugins.pipeline.modeldefinition.model.CredentialsBindingHandler;
+import org.jenkinsci.plugins.plaincredentials.StringCredentials;
+
+import javax.annotation.Nonnull;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Extension
+public class StringCredentialsHandler extends CredentialsBindingHandler<StringCredentials> {
+
+    @Nonnull
+    @Override
+    public List<MultiBinding<StringCredentials>> toBindings(String varName, String credentialsId) {
+        return Collections.<MultiBinding<StringCredentials>>singletonList(new StringBinding(varName, credentialsId));
+    }
+
+    @Nonnull
+    @Override
+    public Class<? extends StandardCredentials> type() {
+        return StringCredentials.class;
+    }
+
+    @Nonnull
+    @Override
+    public List<Map<String, Object>> getWithCredentialsParameters(String credentialsId) {
+        Map<String, Object> map = new HashMap<>();
+        map.put("$class", StringBinding.class.getName());
+        map.put("variable", new EnvVarResolver());
```

Comment:
What `EnvVarResolver` are these classes picking up? The original was a static internal class. I didn't think those were inherited?

Reply:
They are inherited, apparently. I guess we could move `EnvVarResolver` to its own file as well for clarity.

Reply:
Unless someone has started using it already?

Reply:
Yeah, better safe than sorry - let's leave it as is.